### PR TITLE
IBM GHE only: Fix IPv4 address configuration

### PR DESCRIPTION
**Pull request by raviteja-b**
This commit avoids IPv4 DHCP and Static address co-existence 
Disables DHCPv4 while configuring ipv4 static addresses and 
Deletes ipv4 static addresses while enabling DHCPv4.

This behaviour needs to be fixed in the backend Network D-bus design.
This is downstream only change needed untill networkd fixes.

Tested by:
Enable DHCPv4 when ipv4 static addresses configured on the interface.
Configure IPv4 static addresses when DHCPv4 enabled on the interface

### CODE REVIEW

**Review by nkotania — APPROVED** [review]

File: redfish-core/lib/ethernet.hpp
Comment (sunharis) [inline redfish-core/lib/ethernet.hpp:2087]: Can this be moved into createIPv4 method? So that some more error checks are passed? Just before creating the static IP object, we can disable the DHCPv4

File: redfish-core/lib/ethernet.hpp
Comment (sunharis) [inline redfish-core/lib/ethernet.hpp:2047]: This can be moved into the handleDHCPPatch method before invoking setDHCPEnabled

File: redfish-core/lib/ethernet.hpp
Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp:2087]: can be moved into createIPv4 method, but need to change createIPv4 input args.
 and createIPv4 method which was getting used in other places..

File: redfish-core/lib/ethernet.hpp
Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp]: done

File: redfish-core/lib/ethernet.hpp
Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp]: done

File: redfish-core/lib/ethernet.hpp
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:1136]: upstream more common to throw an error if nullptr (it should not be nullptr so something terrible happened if nullptr)

File: redfish-core/lib/ethernet.hpp
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:1115]: A LOG_ERROR here with the errorCode would be useful

File: redfish-core/lib/ethernet.hpp
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:1111]: Pass error_code by reference please. See https://gerrit.openbmc.org/c/openbmc/bmcweb/+/61013

**Comment by gtmills** [comment]
Why GHE only?

**Review by sunharis — APPROVED** [review]

File: redfish-core/lib/ethernet.hpp
Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp:1136]: in this case its just about reading property value to check if its null, it just ignores and continue to check other objects.
same followed in the existing.. I don't anything terrible can happen with bmcweb if this property ptr is null
https://github.com/openbmc/bmcweb/blob/master/redfish-core/lib/ethernet.hpp#L456

File: redfish-core/lib/ethernet.hpp
Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp]: done

File: redfish-core/lib/ethernet.hpp
Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp:1116]: done

**Comment by raviteja-b** [comment]
> Why GHE only?

This workaround for now to turnaround our FVT defects and stabilise 1050 driver, actually networkd design should fixed to handle this configuration, so this won't be acceptable in the upstream for sure.
we have already started discussion with networkd maintainer regarding the same.

File: redfish-core/lib/ethernet.hpp
Comment (gtmills) [inline redfish-core/lib/ethernet.hpp:1186]: Why can't we fix this in the network D-Bus app?

Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp:1186]: networkd design needs to be changed to fix this, have tried to fix but seeing issues in the networkd
we have started discussions for networkd design changes with upstream maintainers.

Comment (gtmills) [reply redfish-core/lib/ethernet.hpp:1186]: I would really prefer we fix this in networkd
bmcweb has enough hacks

Comment (gtmills) [reply redfish-core/lib/ethernet.hpp:1186]: Can we fix in downstream networkd ? Work those upstream?

Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp:1186]: this PR clears at least 5 critical sev1 network defects

Comment (gtmills) [reply redfish-core/lib/ethernet.hpp:1186]: Upstream wouldn’t allow this in bmcweb, I’m struggling why we should allow this downstream in bmcweb. How much effort is the networkd change? A few days or weeks? Can we wait until we make the changes in networkd?

Comment (gtmills) [reply redfish-core/lib/ethernet.hpp:1186]: If you discuss Monday at scrum of scrums overflow and decide this is the correct course, I’m good but I really think we should just fix the right way in networkd now.

Comment (raviteja-b) [reply redfish-core/lib/ethernet.hpp:1186]: Gunnar 
upstream networkd design completely changed and as part 1050 rebase we pulled master code which allows both ipv4 dhcp and static IP co-existence & networkd broken completely, we need to figure out networkd fix yet..
as I mentioned we need to pull this workaround for sometime to unblock fvt

**Pull request closed without merging**
